www/src/components/ThemeSelector.tsx

```diff
@@ -0,0 +1,180 @@
+import { useCurrentTheme } from '../hooks/useCurrentTheme';
+import { useThemeModal } from '../hooks/useThemeModal';
+import { useResetModal } from '../hooks/useResetModal';
+import { useEditMode } from '../hooks/useEditMode';
+import { useThemeContext } from '../hooks/useThemeContext';
+import ViewPanel from './ViewPanel';
+import EditPanel from './EditPanel';
+import ResetThemesModal from './ResetThemesModal';
+import RemoveThemeModal from './RemoveThemeModal';
+import AddThemeModal from './AddThemeModal';
+import EditThemeModal from './EditThemeModal';
+
+export default function ThemeSelector() {
+  const {
+    themes,
+    currentThemeIndex,
+    hasCustomThemes,
+    addTheme,
+    updateTheme,
+    removeTheme,
+    resetThemes,
+    setCurrentTheme,
+  } = useThemeContext();
+
+  const currentTheme = useCurrentTheme();
+
+  const addButtonRef = useRef<HTMLButtonElement | null>(null);
+
+  const handleResetComplete = () => {
+    if (addButtonRef.current) {
+      addButtonRef.current.focus();
+    }
+  };
+
+  const handleAddModalClose = () => {
+    if (addButtonRef.current) {
+      addButtonRef.current.focus();
+    }
+  };
+
+  const {
+    isEditMode,
+    toggleEditMode,
+    editButtonRef,
+    closeButtonRef,
+    radioRefs,
+    handleTransitionEnd,
+  } = useEditMode(currentThemeIndex);
+
+  const handleAddThemeSave = () => {
+    const newThemeIndex = themes.length;
+    const radioRef = radioRefs.current?.[newThemeIndex];
+    if (radioRef) {
+      radioRef.focus();
+    }
+  };
+
+  const {
+    showAddModal,
+    showEditModal,
+    editingTheme,
+    showRemoveConfirm,
+    addFormRef,
+    editFormRef,
+    closeAddModal,
+    closeEditModal,
+    handleAddSaveClick,
+    handleEditSaveClick,
+    handleEditTheme,
+    handleRemoveTheme,
+    handleConfirmRemove,
+    handleCancelRemove,
+    handleAddClick,
+    handleAddTheme,
+    handleEditThemeSave,
+  } = useThemeModal(
+    addTheme,
+    updateTheme,
+    removeTheme,
+    handleAddModalClose,
+    handleAddThemeSave,
+  );
+
+  const {
+    showResetConfirm,
+    openResetConfirm,
+    closeResetConfirm,
+    handleResetConfirm,
+  } = useResetModal(resetThemes, handleResetComplete);
+
+  const handleThemeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const themeIndex = parseInt(value, 10);
+    setCurrentTheme(themeIndex);
+  };
+
+  return (
+    <div>
+      <Stack direction="horizontal" className="justify-content-between">
+        <span className="small">Current theme:</span>
+        {isEditMode ? (
+          <IconButton
+            ref={closeButtonRef}
+            src={Close}
+            alt="Finish editing"
+            size="sm"
+            onClick={toggleEditMode}
+            variant="secondary"
+          />
+        ) : (
+          <Button ref={editButtonRef} size="sm" variant="link" onClick={toggleEditMode}>
+            Edit
+          </Button>
+        )}
```

Comment:
Using "Edit" and :x: for expanding/collapsing the radio set feels a little odd to me. 

https://github.com/user-attachments/assets/4da6da47-67e1-4ef7-b218-2d9fe27c16be

We use :arrow_down_small: and :arrow_up_small: in other places 

https://github.com/user-attachments/assets/d5b37748-f0e8-4c0a-b860-d9116cc3efc4

Would just using a [basic style `Collapsible`](https://paragon-openedx.netlify.app/components/collapsible/#basic-style) here work?